fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/CoordinatorService.java

```diff
@@ -179,6 +189,22 @@ public CompletableFuture<CreateTableResponse> createTable(CreateTableRequest req
                     TableAssignmentUtils.generateAssignment(bucketCount, replicaFactor, servers);
         }
 
+        // validate table properties before creating table
+        validateTableDescriptor(tableDescriptor);
+
+        // before create table in fluss, we may create in lake
+        if (isDataLakeEnabled(tableDescriptor)) {
+            try {
+                getLakeCatalog().createTable(tablePath, tableDescriptor);
+            } catch (TableAlreadyExistException e) {
+                throw new TableAlreadyExistException(
+                        String.format(
+                                "The table %s already exists in %s catalog, please "
+                                        + "first drop the table in %s catalog.",
+                                tablePath, dataLakeFormat, dataLakeFormat));
```

Comment:
Both `drop existed table` and `suggest a new table name` makes sense in this case, the later should be better?

Reply:
I suggest both of them in the message.